docs/source/project_documentation/plugins/generate.md

```diff
@@ -1,13 +1,17 @@
 # Generate
 
-Plugin providing general-purpose content generation and clean-up.
+Plugin providing content generation and clean-up.
 
 ## Generate Command
 
-The `generate` plugin provides a general-purpose content generation capability.  It can be used to generate any text file from a model.
+The `generate` plugin provides a content generation capability for use in user-defined plugins.  It can be used to generate any text file from a model.
+As of right now, it cannot be used cleanly from the command line, and must be called through the `run_generate()` function.
 
-```bash
-aac generate my_model.aac  my_generator.aac
+```python
+from aac.plugins.generate import run_generate
+
+def generate_command(my_model: str, my_generator: str):
+    run_generate(my_model, my_generator)
 ```
 
 ### Arguments
```

Comment:
Can't put a comment on appropriate line. But the next header down is `AaC Plugin File` and it should just be `AaC File`

Reply:
And question, thoughts on if we should make more clarifying statements about the optional flag arguments? So, any argument starting the `--`. This would apply across the commands. I'm not finding commentary anywhere that really covers that aspect. And unless you are used to utilizing a CLI tool, that's not inherently known.

Reply:
What kind of clarifying statements?

Reply:
Should be okay to just tweak the start of each optional argument's statement to be something like:
` The optional argument --code-output`
`The optional argument --evaluate`
etc.

Reply:
Oh I see, so clarifying that the arguments are optional.  Would having the "Arguments" header say "Optional Arguments" instead be sufficient?

Reply:
Issue is that not all of the arguments are optional, but they are all arguments.

Reply:
So could make two sections. `Arguments` which would be required ones like AaC files. And `Optional Arguments` which would be any that start with `--` and provided additional configuration or behavior. 

Which, I think, all of the arguments sections for each plugin are at least already grouped that way with all required ones first then optional following. So should just need to add the `Optional Arguments` before any optional arguments listed out for the plugin.

Reply:
Optional Arguments and non-optional arguments are separated into two sections in the latest commit.